src/main/java/module-info.java

```diff
@@ -0,0 +1,5 @@
+module com.conversantmedia.disruptor {
+    opens com.conversantmedia.util.collection;
```

Comment:
Does `opens` expose these packages at run time only? I've only just started reading about modules so my understanding may be off.

Reply:
Should be both 'test' scope and 'runtime' if your thinking in maven terms.    It exposes them to another package that would use the module jar.   And declare that they 'require' it.

Reply:
Wouldn't that not allow others to write and compile new code that uses this library? Or can that still be done through maven/gradle?

Reply:
No that is all still possible, this just allows projects who are using Java 9 modules to include this dependency.

Reply:
From what I've read, `opens` allows access to public types (and private types through reflection) in a package at **runtime only** for those using java 9 modules. `exports` allows access to public types in a package at compile and runtime. Wouldn't we want `exports` in that case?

Reply:
Thats @evanwht1 that is awesome feedback of course!   I'm not sure if the typical use case for this library is to extend it.   In fact, I'll double check but I'm hoping almost everything is final.   I was able to use this in my project without any trouble, but I'll do another round of testing just in case something is breaking.  
 
What do you think?  Are you suggesting that more of the library should be accessible for extending and creating trial implementations?  As an aside, I have no problem with setting it to extends but always default to the lowest level of permissions.

Reply:
I guess I'm trying to ask 2 different questions. 
1. Do we want to allow reflective access to non-public types and members? (`opens` does that) 
2. Do we want to allow people to directly build applications with the library? The wording of the `opens` directive makes it sound like this library would be used like slf4j implementations, where an application compiles by using some set of interfaces in another module. Then at runtime, the interfaces module finds and uses this library as its implementation. However, I just tried using the library through maven and it seems I can compile just fine so maybe I'm reading the java modules documentation incorrectly.

Since I was able to get my tests working locally, I'm fine with using `opens`.

Reply:
Compiling will continue to work if you are not using the module path to compile. If you do use the module path I would expect the compilation to fail unless the package is exported.